### ISSUE TITLE
Is there any way to run code when speak command finishes?

### ISSUE THREAD

**Issue by ToonTalk** [jmoenig/Snap--Build-Your-Own-Blocks#1408]

**Comment by liam4** [comment]
I'm not sure what Snap*!* library you're using for speaking text, but there *is* an event for your needs: `SpeechSynthesisUtterence`'s [`end`](https://developer.mozilla.org/en-US/docs/Web/Events/end_(SpeechSynthesis)) event. Perhaps listen to that, to continue the script afterwords?

```js
utterence.onend = function() {
  // get the following script to occur.. not 100% sure how to do this :)
  // probably something involving continuations.
}
```

**Comment by ToonTalk** [comment]
Thanks.

I was confused by your question of which library since I didn't load a library. What I just discovered is that when I loaded JSFunctions from Examples that it defined speak and because my project involved other JavaScript blocks I started my project there and then later removed all the sample scripts from my copy of JSFunctions. I didn't realise until just now that the "speak" command that was in the Sounds section was created by JSFunctions and wasn't a standard part of Snap!.

I'll just modify the speak command in JavaScript.
